engine/common/api_helpers/custom_rate_scoped_throttler.py

```diff
@@ -1,55 +1,30 @@
-from rest_framework.throttling import SimpleRateThrottle
+from django.conf import settings
+from ratelimit.utils import _split_rate
+from rest_framework.throttling import UserRateThrottle
 
 
-class CustomRateScopedThrottler(SimpleRateThrottle):
-    """
-    Abstract class to create throttlers with custom amount of seconds and custom scope.
-    The unique cache key will be generated by concatenating the
-    user id of the request, and the scope from get_scope() method.
+class CustomRateUserThrottler(UserRateThrottle):
+    """ """
 
-    Should not be used directly.
-    """
+    def parse_rate(self, rate):
+        "Use django ratelimit format to parse rate, i.e. '30/1m', instead of '30/m'"
+        return _split_rate(rate)
 
-    def __init__(self):
-        self.scope = self.get_scope()
-        self.num_requests, self.duration = self.get_throttle_limits()
-
-    def get_throttle_limits(self):
-        """
-        :return tuple requests/seconds
-        """
-        raise NotImplementedError
+    def allow_request(self, request, view):
+        # Override default rate limit, if organization id is specified in CUSTOM_RATELIMITS
+        custom_ratelimits = settings.CUSTOM_RATELIMITS
+        organization_id = str(request.user.organization_id)
+        if organization_id in custom_ratelimits:
+            self.rate = custom_ratelimits[organization_id]["public_api"]
+        self.num_requests, self.duration = self.parse_rate(self.rate)
 
-    def get_scope(self):
-        """
-        :return ratelimit scope
-        """
-        raise NotImplementedError
+        return super().allow_request(request, view)
```

Comment:
👆 this allows to specify custom ratelimit for public api in settings and env variable

Reply:
same thing here - consider using a dataclass for custom_ratelimits to add type-checks, to not just use dict keys.